contracts/Marketplace.sol

```diff
@@ -6,20 +6,20 @@ import "@openzeppelin/contracts/utils/math/Math.sol";
 import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 import "./Configuration.sol";
 import "./Requests.sol";
-import "./Collateral.sol";
 import "./Proofs.sol";
 import "./StateRetrieval.sol";
 
-contract Marketplace is Collateral, Proofs, StateRetrieval {
+contract Marketplace is Proofs, StateRetrieval {
   using EnumerableSet for EnumerableSet.Bytes32Set;
   using Requests for Request;
 
+  IERC20 public immutable token;
   MarketplaceConfig public config;
 
   MarketplaceFunds private _funds;
   mapping(RequestId => Request) private _requests;
   mapping(RequestId => RequestContext) private _requestContexts;
-  mapping(SlotId => Slot) private _slots;
+  mapping(SlotId => Slot) internal _slots;
```

Comment:
I have changed this in order to have testing functions related to this in the `TestMarketplace`